double_linked_list/double_linked_list.c

```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "double_linked_list.h"
+
+list* list_create(){
+    list* l = (list*)malloc(sizeof(list));
+    l->quantity = 0;
```

Comment:
Both `head` and `tail` pointers are uninitialized. Because of that, most of the functions many functions may fail to work properly because they are based on those uninitialized values...